amazon-kinesis-client/src/test/java/software/amazon/kinesis/config/KCLAppConfig.java

```diff
@@ -19,48 +29,92 @@
 import software.amazon.awssdk.services.kinesis.KinesisAsyncClient;
 import software.amazon.awssdk.services.kinesis.KinesisAsyncClientBuilder;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamSummaryRequest;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamSummaryResponse;
+import software.amazon.awssdk.services.sts.StsAsyncClient;
 import software.amazon.awssdk.utils.AttributeMap;
 import software.amazon.kinesis.common.ConfigsBuilder;
 import software.amazon.kinesis.common.InitialPositionInStream;
 import software.amazon.kinesis.processor.ShardRecordProcessorFactory;
-import software.amazon.kinesis.retrieval.RetrievalConfig;
 
 import java.io.IOException;
 import java.net.Inet4Address;
 import java.net.URISyntaxException;
 import java.net.UnknownHostException;
+import java.time.Duration;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * Default configuration for a producer or consumer used in integration tests.
  * Producer: puts records of size 60 KB at an interval of 100 ms
  * Consumer: streaming configuration (vs polling) that starts processing records at shard horizon
  */
+@Slf4j
 public abstract class KCLAppConfig {
+    public static final String AWS_ACCOUNT_PROFILE_PROPERTY = "awsProfile";
+    public static final String CROSS_ACCOUNT_PROFILE_PROPERTY = "awsCrossAccountProfile";
+    public static final String CROSS_ACCOUNT_CONSUMER_NAME = "cross-account-consumer";
+    public static final String INTEGRATION_TEST_RESOURCE_PREFIX = "KCLIntegrationTest";
 
-    private KinesisAsyncClient kinesisAsyncClient;
+    private String accountIdForConsumer = null;
+    private String accountIdForStreamOwner = null;
+    private List<String> streamNames = null;
+    private KinesisAsyncClient kinesisAsyncClientForConsumer;
+    private StsAsyncClient stsAsyncClientForConsumer;
+    private KinesisAsyncClient kinesisAsyncClientForStreamOwner;
+    private StsAsyncClient stsAsyncClientForStreamOwner;
     private DynamoDbAsyncClient dynamoDbAsyncClient;
     private CloudWatchAsyncClient cloudWatchAsyncClient;
     private RecordValidatorQueue recordValidator;
 
     /**
-     * Name used for test stream and lease tracker table
+     * List of Strings, either stream names or valid stream Arns, to be used in testing. For single stream mode, return
+     * a list of size 1. For multistream mode, return a list of size > 1.
      */
-    public abstract String getStreamName();
+    public abstract List<Arn> getStreamArns();
+
+    public List<String> getStreamNames() {
+        if (this.streamNames == null) {
+            return getStreamArns().stream().map(streamArn ->
+                                          streamArn.toString().substring(streamArn.toString().indexOf("/") + 1))
+                                  .collect(Collectors.toList());
+        } else {
+            return this.streamNames;
+        }
+    }
```

Comment:
i don't think we ever update the `streamNames` field so looks like the second branch would be unreachable here?